docs/docs/smart-contracts/PSP1155/psp1155.md

```diff
@@ -111,14 +114,16 @@ impl MyPSP1155 {
     }
 
     #[ink(message)]
-    pub fn add_type(&mut self, id: Id) {
-        self.registered_ids.insert(id, true);
+    pub fn deny(&mut self, id: Id) {
+        self.denied_ids.insert(id, ());
     }
 
     #[ink(message)]
-    pub fn mint_tokens(&mut self, id: Id, amount: Balance) {
-        assert!(*self.registered_ids.get(&id).unwrap_or(&false));
-        self._mint(id, amount);
+    pub fn mint_tokens(&mut self, id: Id, amount: Balance) -> Result<(), PSP1155Error> {
+        if self.denied_ids.get(&id).is_some() {
+            return Err(PSP1155Error::Custom(String::from("Id is denied")))
+        }
```

Comment:
I think for every example smart contract it's a good idea to create its own error enum. So instead of having `PSP1155Error::Custom` here, I would much prefer having another enum: 
```rust
pub enum PSP1155ExampleError {
  PSP1155Error(PSP1155Error),
  IdDenied,
  MaybeSomeOtherError
}
```

Reply:
We can't do that, because trait should return right "Errors" expected by another contract.  We can't use `PSP1155ExampleError` in `PSP1155` trait.

Reply:
great idea! let's push this paradigm sloooowly but steadily :D